profiling/src/internal/observation/observations.rs

```diff
@@ -58,28 +68,22 @@ impl Observations {
     }
 
     pub fn is_empty(&self) -> bool {
-        self.inner.is_none()
+        self.inner.is_none() ||
+            (self.aggregated_samples_count() == 0 && self.timestamped_samples_count() == 0)
```

Comment:
Because we eagerly initialize the `Observations` now, some of the Profiler tests that checked "when a profile gets created the observations are empty" started failing.